### PR TITLE
Added Dodge and Flee choices

### LINKED ISSUE #107: Add more choices for the player during Battle
The tutorial battle already teaches the player what the Dodge and Flee actions do in combat. Let's add in the real combat too!

### ACCEPTANCE CRITERIA:

- In the beginning of the player's turn, they will be presented with the options that exist in the tutorial battle (Attack, Dodge, Flee)
- Depending on their choice, the appropriate action will be taken: in attack, the battle will continue as is today; in dodge, the enemies will have disadvantage when attacking the character; in flee, a 1d20+Dex mod. dice roll will decide whether the battle is exited or the fleeing fails and the battle continues with the turn of the next enemy (the dice result is considered successful if it is higher than or equal to 12)

### PR DESCRIPTION
**Pull request by DIONYSUS-BASOUNAS**
closes #107 
Added the ability for the main character to choose to Flee the battle or to try to Dodge the next attack

### CODE REVIEW

File: src/battle/battle.cpp [hunk 1]
```diff
@@ -83,65 +84,122 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
...
+									}
+								}
+							}
+							break;
 						}
-					}
-					for (int j = combatants.size() - 1; j >= 0; j--)
-					{
-						if (combatants[j].first->get_hp() == 0)
+						// Dodge option
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:153]: Wrong format? Same with the Flee option comment.

File: src/battle/battle.cpp [hunk 2]
```diff
@@ -83,65 +84,122 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
...
-							combatants.erase(combatants.begin() + j);
+							std::cout << "You have taken a Dodge action" << std::endl;
+							dodge_action = true;
+							break;
+						}
+						// Flee option
+					case 3:
+						{
+							int flee_roll = utils::RollDice(1, 20) +
+								(combatants[i].first->get_ability_score(1) - 10) / 2;
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:164]: Could be `main_character_ptr->get_ability_score` for better readability.

File: src/battle/battle.cpp [hunk 3]
```diff
@@ -83,65 +84,122 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
...
 						}
-					}
-					for (int j = combatants.size() - 1; j >= 0; j--)
-					{
-						if (combatants[j].first->get_hp() == 0)
+						// Dodge option
+					case 2:
 						{
-							combatants.erase(combatants.begin() + j);
+							std::cout << "You have taken a Dodge action" << std::endl;
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:156]:
```suggestion
							std::cout << "You have taken the Dodge action." << std::endl;
```

File: src/battle/battle.cpp [hunk 4]
```diff
@@ -83,65 +84,122 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
...
+				}
 
 				if (main_character_ptr->get_hp() == 0)
 				{
 					break; // Exit the Battle loop
 				}
 			}
+			if (flee == true)
+			{
+				break; // Exit the Battle loop if dodge is successful
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:201]:
```suggestion
				break; // Exit the Battle loop if flee was successful
```

File: src/battle/battle.cpp [hunk 5]
```diff
@@ -150,13 +208,19 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
 	}
 
 	std::cout << "Battle ends!" << std::endl;
-
-	if (main_character_ptr->get_hp() > 0)
+	if (flee == true)
 	{
-		std::cout << "You won the battle!" << std::endl;
+		std::cout << "You have fled the battle" << std::endl;
```
Comment (achilleasmitos) [inline src/battle/battle.cpp:213]: Nice touch! 👍

**Review by achilleasmitos — COMMENTED** [review]
Nice, simple attempt. I do believe that the dodge action should be part of a creature's "conditions", but we can deal with that concept later. Also, the different files for attack eventually need to go away, but it's good as is for now.

File: src/battle/battle.cpp (hunk 1 above)
Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:153]: why is it a wrong format??

Comment (achilleasmitos) [reply src/battle/battle.cpp:153]: Extra tab inserted. Comment should be aligned with the case it is describing.

Comment (DIONYSUS-BASOUNAS) [reply src/battle/battle.cpp:153]: oh

**Review by achilleasmitos — APPROVED** [review]

**Pull request merged**